File: mobile-engage/src/androidTest/java/com/emarsys/mobileengage/testUtil/RandomTestUtils.java
package com.emarsys.mobileengage.testUtil;

import com.emarsys.mobileengage.iam.model.buttonclicked.ButtonClicked;
import com.emarsys.mobileengage.iam.model.displayediam.DisplayedIam;

import java.util.HashMap;
import java.util.Map;
import java.util.Random;
import java.util.UUID;

public class RandomTestUtils {

    public static boolean randomBool() {
        return new Random().nextBoolean();
    }

    public static int randomInt() {
        return new Random().nextInt();
    }

    public static long randomLong() {
        return new Random().nextLong();
    }

    public static String randomString() {
        return UUID.randomUUID().toString();
    }

    public static String randomNumberString() {
        return Long.toString(randomLong());
    }

    public static Object randomMap() {
        Map<String, Object> map = new HashMap<>();
        map.put(randomString(), randomInt());
        map.put(randomString(), randomBool());
        map.put(randomString(), randomString());
        return map;
    }

    public static DisplayedIam randomDisplayedIam() {
        return new DisplayedIam(randomNumberString(), randomLong());
    }

    public static ButtonClicked randomButtonClick() {
        return new ButtonClicked(randomNumberString(), randomNumberString(), randomLong());
    }
}

File: testUtils/src/main/java/com/emarsys/test/util/RandomTestUtils.kt
package com.emarsys.test.util

import java.util.*

object RandomTestUtils {

    @JvmStatic
    fun randomBool(): Boolean {
        return Random().nextBoolean()
    }

    @JvmStatic
    fun randomInt(): Int {
        return Random().nextInt()
    }

    @JvmStatic
    fun randomLong(): Long {
        return Random().nextLong()
    }

    @JvmStatic
    fun randomString(): String {
        return UUID.randomUUID().toString()
    }

    @JvmStatic
    fun randomNumberString(): String {
        return java.lang.Long.toString(randomLong())
    }

    @JvmStatic
    fun randomMap(): Any {
        val map = HashMap<String, Any>()
        map[randomString()] = randomInt()
        map[randomString()] = randomBool()
        map[randomString()] = randomString()
        return map
    }
}
